File: app/src/main/java/com/mirash/passkeeper/view/gesturedetect/HideKeyboardScrollView.java
package com.mirash.passkeeper.view.gesturedetect;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.mirash.passkeeper.tool.Utils;
import com.mirash.passkeeper.view.gesturedetect.tool.SingleTapUpListener;

/**
 * @author Mirash
 */
public class HideKeyboardScrollView extends GestureDetectingScrollView implements SingleTapUpListener {
    public HideKeyboardScrollView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public HideKeyboardScrollView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void init() {
        super.init();
        setSingleTapListener(this);
    }

    @Override
    public boolean onSingleTapUp() {
        Utils.hideKeyboard(getContext());
        return false;
    }
}

File: app/src/main/java/com/mirash/familiar/view/gesturedetect/HideKeyboardScrollView.kt
package com.mirash.familiar.view.gesturedetect

import android.content.Context
import android.util.AttributeSet
import com.mirash.familiar.tool.hideKeyboard
import com.mirash.familiar.view.gesturedetect.tool.SingleTapUpListener

/**
 * @author Mirash
 */
class HideKeyboardScrollView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : GestureDetectingScrollView(context, attrs, defStyleAttr), SingleTapUpListener {

    init {
        setSingleTapListener(this)
    }

    override fun onSingleTapUp(): Boolean {
        hideKeyboard(context)
        return false
    }
}
